src/app/task/task-details/task-details.component.ts

```diff
@@ -81,6 +102,7 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
         type: this.editForm.value.type as string,
```

Comment:
why do you specify the type here? Is there an uncertainty here that demands a type to be specified?